torch/distributed/fsdp/_optim_utils.py

```diff
@@ -160,9 +160,7 @@ def _communicate_optim_state(
                 tensor_buffer = value.new_zeros(*buffer_size)
             dist._all_gather_base(tensor_buffer, value, group=group)
             if to_save:
-                assert hasattr(flat_param, "_orig_size"), \
```

Comment:
The `assert` is not useful. Every `FlatParameter` should call `init_metadata()`, so `_unsharded_size` is guaranteed to be an attribute.